modules/consensus/consensusset.go

```diff
@@ -139,12 +146,21 @@ func New(gateway modules.Gateway, persistDir string) (*ConsensusSet, error) {
 		}
 
 		// Register RPCs
-		gateway.RegisterRPC("SendBlocks", cs.rpcSendBlocks)
-		gateway.RegisterRPC("RelayBlock", cs.rpcRelayBlock) // COMPATv0.5.1
-		gateway.RegisterRPC("RelayHeader", cs.rpcRelayHeader)
-		gateway.RegisterRPC("SendBlk", cs.rpcSendBlk)
+		gateway.RegisterRPC("SendBlocks", cs.threadedRPCSendBlocks)
+		gateway.RegisterRPC("RelayBlock", cs.threadedRPCRelayBlock) // COMPATv0.5.1
+		gateway.RegisterRPC("RelayHeader", cs.threadedRPCRelayHeader)
+		gateway.RegisterRPC("SendBlk", cs.threadedRPCSendBlk)
 		gateway.RegisterConnectCall("SendBlocks", cs.threadedReceiveBlocks)
 
+		// Unregister RPCs on Close.
+		cs.tg.OnStop(func() {
+			cs.gateway.UnregisterRPC("SendBlocks")
+			cs.gateway.UnregisterRPC("RelayBlock") // COMPATv0.5.1
+			cs.gateway.UnregisterRPC("RelayHeader")
+			cs.gateway.UnregisterRPC("SendBlk")
+			cs.gateway.UnregisterConnectCall("SendBlocks")
+		})
+
```

Comment:
This doesn't work. If `Close` is called after the `RegisterRPC` calls but before this `OnStop` call, the `OnStop` call will return immediately without unregistering the RPCs.